zkevm-circuits/src/evm_circuit/execution/error_invalid_opcode.rs

```diff
@@ -0,0 +1,225 @@
+use crate::evm_circuit::execution::ExecutionGadget;
+use crate::evm_circuit::step::ExecutionState;
+use crate::evm_circuit::table::{FixedTableTag, Lookup};
+use crate::evm_circuit::util::common_gadget::RestoreContextGadget;
+use crate::evm_circuit::util::constraint_builder::Transition::{Delta, Same};
+use crate::evm_circuit::util::constraint_builder::{ConstraintBuilder, StepStateTransition};
+use crate::evm_circuit::util::{not, CachedRegion, Cell};
+use crate::evm_circuit::witness::{Block, Call, ExecStep, Transaction};
+use crate::table::CallContextFieldTag;
+use eth_types::Field;
+use gadgets::util::Expr;
+use halo2_proofs::circuit::Value;
+use halo2_proofs::plonk::Error;
+
+/// Gadget for invalid opcodes. It verifies by a fixed lookup for
+/// ResponsibleOpcode.
+#[derive(Clone, Debug)]
+pub(crate) struct ErrorInvalidOpcodeGadget<F> {
+    opcode: Cell<F>,
+    rw_counter_end_of_reversion: Cell<F>,
+    restore_context: RestoreContextGadget<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for ErrorInvalidOpcodeGadget<F> {
+    const NAME: &'static str = "ErrorInvalidOpcode";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::ErrorInvalidOpcode;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
```

Comment:
so invalid op code can not be larger than 255 ?

Reply:
evm run definitely never meet > 255 byte code since it use `byte` type for op code id. but for circuit, adversary can assign any invalid value for `opcode` cell I think.

Reply:
Fixed to add a `LtGadget` to constrain `opcode` cell must be less than `256` (to avoid hacking assign) in commit [537748b](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1089/commits/537748bc9aca466581867274a90581d93c63806c).

Reply:
Yes. I see. It seems that `ResponsibleOpcode` lookup should fail (make this circuit failed) if `opcode` is assigned to a value greater than 255.

Reverted to delete the `LtGadget` in 915e8698bcf157f5fc5cabc01c62ed71ea34998b.